### PR TITLE
#10289 Fix incorrect use of _T in react signature

### LINKED ISSUE #10289: Type hint for twisted.internet.task.react is incorrect?
|[euresti's avatar](https://github.com/euresti)| @euresti reported|
|-|-|
|Trac ID|trac#10289|
|Type|enhancement|
|Created|2022-01-08 00:04:15Z|

It's currently typed as:
```
def react(
    main: Callable[
        ...,
        Union[Deferred[_T], Coroutine["Deferred[_T]", object, _T]],
    ],
    argv: Iterable[_T] = (),
    _reactor: Optional[IReactorCore] = None,
) -> NoReturn:
```
But I'm not sure argv should be `_T` since its not related to the return value of the callable.
 

(I'm not sure if there's a way to express in types that main should be take a reactor + whatever is in argv)

<details><summary>Searchable metadata</summary>

```
trac-id__10289 10289
type__enhancement enhancement
reporter__euresti euresti
priority__normal normal
milestone__None None
branch__ 
branch_author__ 
status__new new
resolution__None None
component__core core
keywords__None None
time__1641600255996109 1641600255996109
changetime__1641600255996109 1641600255996109
version__None None
owner__None None

```
</details>

### PR DESCRIPTION
**Pull request by twm**
## Scope and purpose

Fixes #10289.

Small PR to give the new process a spin.

I briefly poked at using [`ParamSpec`](https://docs.python.org/3/library/typing.html#typing.ParamSpec) to better type the callable but it looks like our MyPy is too old and I don't feel like shaving that yak right now. This addresses the immediate issue.

### CODE REVIEW

**Comment by twm** [comment]
Let's give this a try: "review required"

**Comment by twm** [comment]
Uh, "please review".

**Comment by twm** [comment]
@chevah-robot please review :pray:

File: src/twisted/internet/task.py [hunk 1]
```diff
@@ -900,10 +900,11 @@ def react(
...
     The following demonstrates the signature of a C{main} function which can be
     used with L{react}::
-          async def main(reactor, username, password):
-              return "ok"
 
-          task.react(main, ("alice", "secret"))
+      async def main(reactor, username, password):
+          return "ok"
+
+      task.react(main, ("alice", "secret"))
```
Comment (twm) [inline src/twisted/internet/task.py:907]: View this here: https://twisted--11598.org.readthedocs.build/en/11598/api/twisted.internet.task.html

**Comment by twm** [comment]
please-review

**Comment by twm** [comment]
review-please

**Review by adiroiban — APPROVED** [review]
Looks good. Thanks also for the docs fix https://twisted--11598.org.readthedocs.build/en/11598/api/twisted.internet.task.html#react

----

Sorry for the review confusion .

I will check to get the review bot up and runnig.

**Comment by adiroiban** [comment]
I have enabled the auto-merge. I hope all is ok and this will land.

Also, thanks for the dev docs drive-by.

**Comment by twm** [comment]
Thanks @adiroiban!

**Pull request merged**